File: MySharedBlazorLib/WindowFunctions.cs
using System;
using Microsoft.AspNetCore.Blazor.Browser.Interop;

namespace MySharedBlazorLib
{
    public class WindowFunctions
    {
        public static void Alert(string msg)
        {
            RegisteredFunction.Invoke<string>("ShowAlert", msg);
        }

        public static string AlertShowCallback(string msg)
        {
            Console.WriteLine(msg);
            return "message written";
        }
    }
}

File: MySharedBlazorLib/content/windowFunctions.js
// This file is to show how a library package may provide JavaScript interop features
// wrapped in a .NET API

Blazor.registerFunction('ShowAlert', (message) => {
    alert(message);
    Blazor.invokeDotNetMethod(
        {
            type: {
                assembly: "MySharedBlazorLib",
                name: "MySharedBlazorLib.WindowFunctions"
            },
            method: {
                name: "AlertShowCallback"
            }
        },
        "calling dotnet method"
    );
});

